homeassistant/components/mqtt/device_tracker/schema_yaml.py

```diff
@@ -78,16 +83,20 @@ def _entry_unload(*_: Any) -> None:
     for dev_id, topic in devices.items():
 
         @callback
-        def async_message_received(msg, dev_id=dev_id):
+        def async_message_received(msg: ReceiveMessage, dev_id: str = dev_id) -> None:
             """Handle received MQTT message."""
             if msg.payload == payload_home:
                 location_name = STATE_HOME
             elif msg.payload == payload_not_home:
                 location_name = STATE_NOT_HOME
             else:
+                assert isinstance(msg.payload, str)
                 location_name = msg.payload
```

Comment:
Is there not a way to avoid this assert?
Maybe by using `location_name = str(msg.payload)`